### PR TITLE
issue 43 koin integration

### PR DESCRIPTION
**Pull request by ben-mathu**
## What is the Purpose?
Koin integration and updating the README.md file to include documentation for how do go about creating new dependencies.

## What was the approach?
Based on the project's architectural principles. Also each module required koin dependencies.

## Mentions?
@michaelbukachi 

## Issue(s) affected?
#43

### CODE REVIEW

**Comment by michaelbukachi** [comment]
Hey @ben-mathu Thanks for your PR! Great work.
A couple of minor changes. 
Please remove Any room related code besides the setup class. We are really trying to organize PRs and tasks involved in those PRs for easier tracking and to avoid scope creep. The following classes are not needed for this PR:
- All room related classes/interfaces except for `DroidConDB`
- All logic added in the `auth` module
- All repositories/data sources

@wangerekaharun can you also review this?

**Comment by ben-mathu** [comment]
@michaelbukachi what about the api service interface in network module

**Comment by michaelbukachi** [comment]
That too should be removed. This PR should entail koin and the bare minimum modules.

File: network/src/main/java/com/android254/droidconKE2020/network/di/Url.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.android254.droidconKE2020.network.di
+
+/**
+ * 18/03/20
+ * @author bernard
+ */
+object Url {
+    const val BASE_URL = "https://example.com"
```
Comment (JabezNzomo99) [inline network/src/main/java/com/android254/droidconKE2020/network/di/Url.kt]: Hey @ben-mathu what do you think about having one object called Constants where we can put all the constants?

Comment (ben-mathu) [reply network/src/main/java/com/android254/droidconKE2020/network/di/Url.kt]: Yeah, but where will it sit? Should it be in the core module.

Comment (JabezNzomo99) [reply network/src/main/java/com/android254/droidconKE2020/network/di/Url.kt]: If the constants are only being used by one feature, i'd place it inside that feature

Comment (ben-mathu) [reply network/src/main/java/com/android254/droidconKE2020/network/di/Url.kt]: Okay got it.

**Comment by michaelbukachi** [comment]
Looks good over here. @JabezNzomo99 If it's alright with you, you can merge the PR.

**Comment by wangerekaharun** [comment]
[Wire up koin in all modules](https://trello.com/c/ETWnaSTI/26-wire-up-koin-in-all-modules)

**Pull request merged**